database/sqlite/sqlite_health.c

```diff
@@ -0,0 +1,931 @@
+    if (unlikely(rc != SQLITE_OK))
+        goto bind_fail;
+
+    if (cfg->p_db_lookup_after) {
+        param++;
+        rc = sqlite3_bind_text(res, 29, cfg->p_db_lookup_dimensions, -1, SQLITE_STATIC);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+
+        param++;
+        rc = sqlite3_bind_text(res, 30, cfg->p_db_lookup_method, -1, SQLITE_STATIC);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+
+        param++;
+        rc = sqlite3_bind_int(res, 31, cfg->p_db_lookup_options);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+
+        param++;
+        rc = sqlite3_bind_int(res, 32, cfg->p_db_lookup_after);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+
+        param++;
+        rc = sqlite3_bind_int(res, 33, cfg->p_db_lookup_before);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+    } else {
+        param++;
+        rc = sqlite3_bind_null(res, 29);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+        param++;
+        rc = sqlite3_bind_null(res, 30);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+        param++;
+        rc = sqlite3_bind_null(res, 31);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+        param++;
+        rc = sqlite3_bind_null(res, 32);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+        param++;
+        rc = sqlite3_bind_null(res, 33);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+    }
+
+    param++;
+    rc = sqlite3_bind_int(res, 34, cfg->p_update_every);
+    if (unlikely(rc != SQLITE_OK))
+        goto bind_fail;
+
+    rc = execute_insert(res);
+    if (unlikely(rc != SQLITE_DONE))
+        error_report("Failed to store alert config, rc = %d", rc);
+
+    rc = sqlite3_reset(res);
+    if (unlikely(rc != SQLITE_OK))
+        error_report("Failed to reset statement in alert hash_id store function, rc = %d", rc);
+
+    return 0;
+
+    bind_fail:
+    error_report("Failed to bind parameter %d to store alert hash_id, rc = %d", param, rc);
+    rc = sqlite3_reset(res);
+    if (unlikely(rc != SQLITE_OK))
+        error_report("Failed to reset statement in alert hash_id store function, rc = %d", rc);
+    return 1;
+}
+
+int alert_hash_and_store_config(
+    uuid_t hash_id,
+    struct alert_config *cfg)
+{
+    EVP_MD_CTX *evpctx;
+    unsigned char hash_value[EVP_MAX_MD_SIZE];
+    unsigned int hash_len;
+    evpctx = EVP_MD_CTX_create();
+    EVP_DigestInit_ex(evpctx, EVP_sha256(), NULL);
+
+    cfg->alarm ? EVP_DigestUpdate(evpctx, cfg->alarm, strlen(cfg->alarm)) : EVP_DigestUpdate(evpctx, "NULL", 4);
+    cfg->template_key ? EVP_DigestUpdate(evpctx, cfg->template_key, strlen(cfg->template_key)) : EVP_DigestUpdate(evpctx, "NULL", 4);
+
+    cfg->os ? EVP_DigestUpdate(evpctx, cfg->os, strlen(cfg->os)) : EVP_DigestUpdate(evpctx, "NULL", 4);
```

Comment:
don't want to be nit picking here but would still prefer to add single byte that is 0 here instead of `NULL` to not create clash in case the pointer != NULL and case where there is a string that has value "NULL".

Additionally for readability I would macroize this or do it as static function

Reply:
Meaning if I make my hostname "NULL" the hash should be different from case where there is no hostname or we couldnt read it or dont know it (pointer is NULL)

Reply:
e.g.:
```
uint_8t x = 0
EVP_DigestUpdate(evpctx, &x, 1);
```